### PR TITLE
(issue #3) Show metadata for packages in the tree

### PR DESCRIPTION
**Pull request by ajroetker**
This commit adds the ability to pass in a format string for displaying
the version in the tree.

This commit also adds the ability to pass in `--charset none` which will
simply list the dependencies instead of displaying them in a graph.

### CODE REVIEW

**Comment by ajroetker** [comment]
Meant to address https://github.com/sfackler/cargo-tree/issues/3

**Comment by ajroetker** [comment]
@sfackler Was this closer to what you had in mind?

Example:

```
[maint/show_metadata_in_tree][~/Projects/rust/cargo-tree]$ ./target/debug/cargo-tree tree --format '{p} {r} {l}' --charset none
cargo-tree v0.4.2 (file:///Users/aroetker/Projects/rust/cargo-tree) https://github.com/sfackler/cargo-tree MIT/Apache-2.0
cargo v0.10.0 https://github.com/rust-lang/cargo MIT/Apache-2.0
advapi32-sys v0.1.2 https://github.com/retep998/winapi-rs MIT
winapi v0.2.6 https://github.com/retep998/winapi-rs MIT
crates-io v0.2.0 https://github.com/rust-lang/cargo MIT/Apache-2.0
<clipped>
```

and

```
[maint/show_metadata_in_tree][~/Projects/rust/cargo-tree]$ ./target/debug/cargo-tree tree --format '{p} {r} {l}'
cargo-tree v0.4.2 (file:///Users/aroetker/Projects/rust/cargo-tree) https://github.com/sfackler/cargo-tree MIT/Apache-2.0
├── cargo v0.10.0 https://github.com/rust-lang/cargo MIT/Apache-2.0
│   ├── advapi32-sys v0.1.2 https://github.com/retep998/winapi-rs MIT
│   │   └── winapi v0.2.6 https://github.com/retep998/winapi-rs MIT
│   ├── crates-io v0.2.0 https://github.com/rust-lang/cargo MIT/Apache-2.0
<clipped>
```

**Comment by sfackler** [comment]
Yeah, this looks good!

`--charset none` seems like a bit of a stretch on the term "charset" 😄 . Maybe it would make more sense to have a `--no-indent` flag? I'm not sure what the right way to do default-on flags with docopt is though.

File: src/main.rs [hunk 1]
```diff
@@ -285,42 +315,57 @@ fn print_tree<'a>(package: &'a PackageId,
...
+            let c = if last_continues {
+                symbols.tee
+            } else {
+                symbols.ell
+            };
+            print!("{0}{1}{1} ", c, symbols.right);
+        }
+    }
+
+    let metadata = graph.node_metadata.get(package).unwrap().clone();
```
Comment (sfackler) [inline src/main.rs:399]: Can you move this message format logic out to a separate function? As we add more substitution patterns it'll get more complicated.

Comment (ajroetker) [inline src/main.rs:399]: Moved that logic [here](https://github.com/sfackler/cargo-tree/pull/5/files#diff-639fbc4ef05b315af92b4d836c31b023R349)

**Comment by ajroetker** [comment]
@sfackler I removed the `--charset none` in favor of the `--no-indent` flag :) much better now.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**